src/components/UserSelector.tsx

```diff
@@ -1,35 +1,91 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+
+type Props = {
+  users: User[];
+  select: User | null;
+  onSelect: React.Dispatch<React.SetStateAction<User | null>>
+  getUserPosts: (userId: number) => void;
+  setIsCommentListHidden: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  select,
+  onSelect,
+  getUserPosts,
+  setIsCommentListHidden,
+}) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const selectRef = useRef(null);
+
+  const handleSelectUser = (user: User) => {
+    onSelect(user);
+    setIsVisible(false);
+    getUserPosts(user.id);
+    setIsCommentListHidden(true);
+  };
+
+  const handleClick = (ev: MouseEvent) => {
+    if (ev.target !== selectRef.current && isVisible) {
+      setIsVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick);
+
+    return () => {
+      document.removeEventListener('click', handleClick);
+    };
+  }, [isVisible]);
 
-export const UserSelector: React.FC = () => {
   return (
     <div
       data-cy="UserSelector"
       className="dropdown is-active"
+      ref={selectRef}
     >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setIsVisible(true)}
         >
-          <span>Choose a user</span>
+          <span>
+            {select
+              ? select.name
+              : 'Choose a user'}
+          </span>
 
           <span className="icon is-small">
             <i className="fas fa-angle-down" aria-hidden="true" />
           </span>
         </button>
       </div>
 
-      <div className="dropdown-menu" id="dropdown-menu" role="menu">
-        <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">Leanne Graham</a>
-          <a href="#user-2" className="dropdown-item is-active">Ervin Howell</a>
-          <a href="#user-3" className="dropdown-item">Clementine Bauch</a>
-          <a href="#user-4" className="dropdown-item">Patricia Lebsack</a>
-          <a href="#user-5" className="dropdown-item">Chelsey Dietrich</a>
+      {isVisible && (
+        <div className="dropdown-menu" id="dropdown-menu" role="menu">
+          <div className="dropdown-content">
+            {users.map(user => (
+              <a
```

Comment:
Avoid using anchor tags for user interactions other than navigation. Use button tags for better accessibility and to prevent page jumps on click.